api-client/src/Endpoint.ts

```diff
@@ -80,6 +82,12 @@ class Endpoint<P, T> {
       })
     })
 
+    // separate check for NotFoundErrors: response is not ok, but ResponseError should not be thrown
```

Comment:
I think the comment does not really add additional information so I'd just remove it.